### PR TITLE
Enable APY e2e tests

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Enable APY e2e tests and resolve issues.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

n/a

### CODE REVIEW

**Review by Karrenbelt — DISMISSED** [review]

**Comment by Adamantios** [comment]
I have detected the reason for the `NETWORK_ERROR` event that gets returned from the `CollectHistoryRound` and `CollectLatestHistoryBatchRound`. This happens when we try to get the ETH price for a specific fantom block:

https://github.com/valory-xyz/open-autonomy/blob/5e3232eb4128330c43b16bdb6cd549fa760f57f3/packages/valory/skills/apy_estimation_abci/behaviours.py#L279-L287

The error that is returned from SpookySwap's subgraph is the following:
```
"Failed to decode `block.number` value: `subgraph QmPJbGjktGa7c4UYWXvDRajPxpuJBSZxeQK5siNT3VpthP has only indexed up to block number 41467188 and data for block number 41516654 is therefore not yet available`"
```

This might not be a big problem if the APY oracle runs with large intervals, e.g., 1 day. However, this is problematic for the e2e tests and may potentially be a problem if we decide to run the oracle for every hour in the future. Should we find another way to retrieve the ETH price at a specific block?

File: .github/workflows/main_workflow.yml [hunk 1]
```diff
@@ -156,7 +156,6 @@ jobs:
           submodules: recursive
       - uses: actions/setup-python@master
         with:
-          submodules: recursive
```
Comment (Adamantios) [inline .github/workflows/main_workflow.yml:159]: This was incorrect.

Comment (dvilelaf) [inline .github/workflows/main_workflow.yml:159]: Is this not neccesary?

File: tests/test_skills/test_apy_estimation_abci/test_apis.py [hunk 2]
```diff
@@ -97,7 +102,44 @@ def test_eth_price(spooky_specs: SpecsType, eth_price_usd_q: str) -> None:
         res = make_request(api.get_spec(), eth_price_usd_q)
         eth_price = ast.literal_eval(api.process_response(DummyMessage(res.content))[0]["ethPrice"])  # type: ignore
 
-        assert isinstance(eth_price, float)
+        assert eth_price == 0.4183383786296383
```
Comment (dvilelaf) [inline tests/test_skills/test_apy_estimation_abci/test_apis.py:105]: We should have that price defined in a constant.

File: tests/test_skills/test_apy_estimation_abci/test_apis.py [hunk 3]
```diff
@@ -97,7 +102,44 @@ def test_eth_price(spooky_specs: SpecsType, eth_price_usd_q: str) -> None:
...
+    def test_eth_price_non_indexed_block(
+        spooky_specs: SpecsType, eth_price_usd_q: str
+    ) -> None:
+        """Test SpookySwap's eth price request from subgraph, when the requesting block has not been indexed yet."""
+        spooky_specs["response_key"] = "errors"
+        api = SpookySwapSubgraph(**spooky_specs)
+
+        # replace the block number with a huge one, so that we get a not indexed error
+        current_number = "3830367"
+        largest_acceptable_number = "2147483647"
```
Comment (dvilelaf) [inline tests/test_skills/test_apy_estimation_abci/test_apis.py:117]: Maybe use constants here as well

**Review by dvilelaf — COMMENTED** [review]
Looks good, just some small comments.

File: .github/workflows/main_workflow.yml (hunk 1 above)
Comment (Adamantios) [reply .github/workflows/main_workflow.yml:159]: No, this is probably a copy-paste from the checkout.

d5b38a5

File: tests/test_skills/test_apy_estimation_abci/test_apis.py (hunk 2 above)
Comment (Adamantios) [reply tests/test_skills/test_apy_estimation_abci/test_apis.py]: Done in 65763b23c6c6c3d5c2f4ab662999c8d303602b72.

File: tests/test_skills/test_apy_estimation_abci/test_apis.py (hunk 3 above)
Comment (Adamantios) [reply tests/test_skills/test_apy_estimation_abci/test_apis.py]: Done in 65763b23c6c6c3d5c2f4ab662999c8d303602b72.

**Review by dvilelaf — APPROVED** [review]
LGTM. There are some merge conflicts.

**Review by dvilelaf — COMMENTED** [review]
Apart from the known problematic tests with registration i don't see a problem with this PR

**Pull request merged**